transformer_engine/pytorch/ops/basic/basic_linear.py

```diff
@@ -506,19 +560,31 @@ def _functional_forward(
         x_async = None
         if with_fp8_compute:
             kwargs = dict(
+                accumulate=accumulate_into_out,
                 out=y,
                 bias=b,
                 use_bias=(b is not None),
             )
             if with_fp8_output:
-                fp8_meta_key = FP8GlobalStateManager.get_meta_tensor_key(
-                    forward=y._fp8_meta_forward,
-                )
+                if y._fp8_meta is None:
+                    # Hackily create FP8TensorMeta if needed
+                    fp8_meta = FP8TensorMeta()
+                    fp8_meta.scale = y._scale_inv.reciprocal()
+                    fp8_meta.amax_history = torch.empty(1, 1, dtype=torch.float32, device=device)
```

Comment:
That should be torch.zeros actually - we do assume amax contains zero at the beginning.

Reply:
That said, what would be the situation when this path would be correct and right thing to do?

Reply:
We don't use the amax afterwards, so zeroing is an extra kernel call. I think the right behavior would be to pass in empty tensors and to entirely avoid computing the amax.